| Variable | Descripción / Traducción | Tipo de Dato | Importancia |
| :--- | :--- | :--- | :--- |
| **`user_id`** | Identificador único del usuario | Numérico Entero (`int`) | Permite trackear, indexar y asegurar que no haya registros duplicados. |
| **`age`** | Edad de la persona | Numérico Entero (`int`) | Es clave para segmentar la audiencia y analizar qué rangos etarios consumen más. |
| **`subscription_plan`** | Plan de suscripción actual (Básico, Estándar, Premium) | Texto / Categórico (`string`) | Define directamente el nivel de ingresos y el tipo de cuenta del usuario. |
| **`monthly_watch_time_mins`** | Tiempo de reproducción mensual en minutos | Numérico Decimal (`float`) | Mide el nivel de enganche (engagement) y uso real de la plataforma. |
| **`country`** | País de residencia del usuario | Texto / Categórico (`string`) | Fundamental para análisis geográficos y distribución de contenido por región. |
| **`favorite_genre`** | Género cinematográfico o televisivo favorito | Texto / Categórico (`string`) | Ayuda a entender las preferencias de consumo para mejorar las recomendaciones. |
| **`last_login_date`** | Última fecha en la que el usuario inició sesión | Fecha (`date` / `string`) | Permite identificar usuarios inactivos o en riesgo de abandonar el servicio (churn). |
| **`customer_support_tickets`** | Cantidad de tickets de soporte técnico generados | Numérico Entero (`int`) | Indica el nivel de insatisfacción o problemas técnicos que experimenta el usuario. |

Se incluye una tabla con el significado de cada columna, tipo de dato y su utilidad, para facilitar la interpretación del dataset

In [2]:
# Importar las librerías estándar de la clase
import pandas as pd
import numpy as np

# Cargar el dataset 
df = pd.read_json('../data/raw/streaming_users_dirty.json')

# Primer vistazo a los datos
print(df.head())

   user_id  age subscription_plan  monthly_watch_time_mins   country  \
0    10000   39          Estándar                    805.8    Brasil   
1    10001   37          Estándar                   1173.4  Colombia   
2    10002   28            Básico                    401.0  Colombia   
3    10003   43            Básico                     62.4   Uruguay   
4    10004   51            Básico                    477.8      Perú   

  favorite_genre last_login_date  customer_support_tickets  
0          Crime      2025-03-04                        99  
1          Crime      2019-04-02                         2  
2          Crime      2018-04-13                         0  
3       Thriller      2021-01-31                         0  
4       Thriller      2020-09-30                         1  


head( ) permite ver los primeros datos del dataset para lograr ver si se cargó correctamente y tener una impresión sobre el contenido

In [3]:
# Mostrar las dimensiones del dataset (Filas y Columnas)
print("Dimensiones del dataset original (Filas, Columnas):")
print(df.shape)

Dimensiones del dataset original (Filas, Columnas):
(8160, 8)


Se logra apreciar que hay 8160 filas y 8 columnas

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   str    
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   str    
 5   favorite_genre            7920 non-null   str    
 6   last_login_date           7840 non-null   str    
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), str(4)
memory usage: 510.1 KB


Se muestra el tipo de dato de cada columna y la cantidad de valores no nulos, se observa que hay algunas variables que cuentan con valores faltantes

In [9]:
# Mostrar la cantidad de valores nulos y su porcentaje
nulos = df.isnull().sum()

porcentaje = (nulos/len(df))*100

pd.DataFrame({
    "Cantidad": nulos,
    "Porcentaje": porcentaje.round(2)
})

,Cantidad,Porcentaje
user_id,0,0.00
age,0,0.00
subscription_plan,0,0.00
monthly_watch_time_mins,193,2.37
country,0,0.00
favorite_genre,240,2.94
last_login_date,320,3.92
customer_support_tickets,0,0.00


Se logra apreciar que hay valores nulos en 3 columnas 

In [6]:
df.describe().round(2)

,user_id,age,monthly_watch_time_mins,customer_support_tickets
count,8160.00,8160.00,7967.00,8160.00
mean,13995.43,34.10,1107.35,1.80
std,2310.81,14.51,5310.44,11.33
min,10000.00,-5.00,-120.00,-1.00
25%,11987.75,25.00,489.20,0.00
50%,13998.50,33.00,757.40,1.00
75%,15997.25,42.00,1045.70,1.00
max,17999.00,150.00,99999.00,150.00


El resumen que nos da la función describe( ) muestra anomalías, por ejemplo, en la columna age presenta un valor mínimo de -5 y un valor máximo de 150, valores poco realistas para la edad de una persona

In [11]:
#Valores únicos
for col in ["country","subscription_plan"]:
    print(col)
    print(df[col].unique())

country
<StringArray>
[    'Brasil',   'Colombia',    'Uruguay',       'Perú',      'Chile',
  'Argentina',     'México',     'Brazil',     'brasil',     'méxico',
      'chile',    'uruguay',        'MEX',        'ARG',   'colombia',
        'COL',     'Mexico',        'URY',     'Chile ',       'Peru',
       'perú',  'argentina',        'PER', 'Argentina ',        'BRA',
        'CHL']
Length: 26, dtype: str
subscription_plan
<StringArray>
[ 'Estándar',    'Básico',   'Premium',       'Std',  'estandar',    'basico',
    'básico',  'Premium ',   'premium',   'Premiun',    'BASICO',  'STANDARD',
     'Basic', 'Estándar ',   'PREMIUM']
Length: 15, dtype: str


Aquí se logra ver que hay muchos errores ya que hay datos escritos de diferentes maneras dando así errores si queremos ver por ejemplo cuantos usuarios hay en Brasil.

In [12]:
# Valores duplicados
print("Cantidad de registros duplicados:")
print(df.duplicated().sum())

Cantidad de registros duplicados:
126


Se observaron 126 registros duplicados en el siguiente notebook se verá que decisión se tomará con respecto a ellos